packages/common/src/interceptors/test/geoip.interceptor.spec.ts

```diff
@@ -0,0 +1,93 @@
+import { GeoIPInterceptor } from '../geoip.interceptor';
+import {
+  ExecutionContext,
+  HttpStatus,
+  CallHandler,
+  HttpException,
+  InternalServerErrorException,
+  Logger,
+} from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { HttpService } from '@nestjs/axios';
+import { Test, TestingModule } from '@nestjs/testing';
+import { Observable, of } from 'rxjs';
+
+describe('GeoIPInterceptor', () => {
+  let geoIPInterceptor: GeoIPInterceptor;
+  let httpService: HttpService;
+  let logger: Logger;
+
+  beforeEach(async () => {
+
+    const mockHttpService = {
+      axiosRef: {
+        get: jest.fn(),
+      },
+    };
+
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        {
+          provide: HttpService,
+          useValue: mockHttpService,
+        },
+        ConfigService,
+        {
+          provide: Logger,
+          useValue: {
+            verbose: jest.fn(),
+            error: jest.fn(),
+            log: jest.fn(),
+          },
+        },
+        {
+          provide: GeoIPInterceptor,
+          useFactory: () => new GeoIPInterceptor({
+            countries: ['India', 'United States'],
+            cities: ['Mumbai', 'New York'],
+            coordinates: [{ lat: 35.6897, lon: 139.6895 }], // Tokyo
+            geofences: [{ lat: 51.5074, lon: -0.1278, radius: 50 }], // London, UK
```

Comment:
could you document how do these params work together when more than one are defined?


will it work if the user sends in the location for Mumbai as a geofence? if we have already allowed for "India" and "United States" in countries then why do we even need to add cities? Will it allow for requests from tokyo and london because the coordinates and geofences allow for them? seems very confusing.

Reply:
> will it work if the user sends in the location for Mumbai as a geofence?
Yes
> if we have already allowed for "India" and "United States" in countries then why do we even need to add cities?
I had added mumbai and new york as an example, we could add cities of another countries to work with that specific city.
> Will it allow for requests from tokyo and london because the coordinates and geofences allow for them?
Yes

**Every paramater is independent with each other.**
**The request is allowed from the list of given countries, cities, coordinates and geofences.**

Reply:
Do you think this independence is good? if I have allowed the request for country India and I have added coordinates for Mumbai in the config and then I send lat long from Delhi to access, will my request be allowed to pass through or will it be rejected?

Reply:
The thing is as India is a country so if we add any city which belongs to India it will allow the request. 
All the params work as `||`
I think I should have added a city which doesn't belong to the specified(allowed) country.